citar.el

```diff
@@ -834,12 +821,6 @@ repeatedly."
              (propertize (citar--get-template 'suffix) 'face 'citar)))
    (error "No template for \"%s\" - check variable 'citar-templates'" template-name)))
 
-(defun citar--all-keys ()
```

Comment:
On your question, FYI, this was just a function used in the oc-basic completion table function.

And you added an equivalent to the first one, though without the filter arg.

Reply:
Yeah, it occurred to me that in general, to do filtering properly, you have to construct the entire hash table and then remove the entries you don't want. Otherwise there can be complications with multiple bibliographies, where a key in an earlier bibliography overrides a later one. If you naively filter the bibliographies and then merge them (or equivalent) then you can end up filtering out the first instance of that key and not the second, which then incorrectly gets "unshadowed". So you pretty much have to first merge all the bibliographies and _then_ filter.

That being the case, it's probably easier to just have the function that returns all the keys/items as a hash table, and use `map-filter` to filter them:
```
(map-filter (citar-has-note) (citar--get-entries)) ; Returns an alist
```

Reply:
That particular code won't work though, since `citar--get-entries` returns a hash table.

Reply:
The functions in `map.el` do indeed work on hash tables (and other kinds of associative data structures). It's the counterpart of `seq.el`, whose functions work generically on sequences.

Reply:
BTW, WDYT of making the get functions (at least for the data) public again?

We want people to feel comfortable using them; right?

Reply:
> The functions in map.el do indeed work on hash tables (and other kinds of associative data structures.

Hmm ... I got an error.

Reply:
I just ran it and it works fine, so the error must be something else.

> WDYT of making the get functions (at least for the data) public again?

Yeah, I think these functions are probably going to pretty stable (and should be easy to keep updated if the implementation changes).